### PR TITLE
UI for choice of measurement units

### LINKED ISSUE #7: UI for choice of measurement units
Requirements:
- Add a select with a list of possible measument temperature units
- If a new value is set, refetch weather data
- By default, temperature should be displayed in celcius units

### PR DESCRIPTION
**Pull request by coding-vibe**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
 import { useState } from 'react';
 import LocationAutocomplete from 'components/LocationAutocomplete';
 import WeatherWidget from 'components/WeatherWidget';
+import MeasurementSelect from 'components/MeasurementSelect';
 import Location from 'types/location';
 
 const LOCATION_AUTOCOMPLETE = 'location-select';
+const MEASUREMENT_SELECT = 'unit-select';
```
Comment (singvarr) [inline src/App.tsx]: Replace word measurement by temperatureUnits everywhere

File: src/components/MeasurementSelect/MeasurementSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <FormControl css={classes.select}>
+      <InputLabel id='helper-label'>Temperature unit</InputLabel>
+      <Select
+        labelId='helper-label'
+        id={id}
+        value={unit}
+        label='Temperature unit'
+        onChange={handleChange}>
+        <MenuItem value='standard'>Kelvin, &deg;&#8490;</MenuItem>
```
Comment (singvarr) [inline src/components/MeasurementSelect/MeasurementSelect.tsx]: Move options to const

File: src/components/MeasurementSelect/MeasurementSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import * as classes from './styles';
+
+interface Props {
+  unit: string;
+  setUnit: (value: string) => void;
```
Comment (singvarr) [inline src/components/MeasurementSelect/MeasurementSelect.tsx]: make `unit` and `value` stricter. It accepts not any string by a fixed range of values

File: src/components/MeasurementSelect/styles.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { css } from '@emotion/react';
+
+export const select = css`
+  margin: 1;
```
Comment (singvarr) [inline src/components/MeasurementSelect/styles.ts]: External geometry should be passed via props.
We need to talk about it

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 5]
```diff
@@ -77,9 +79,9 @@ export default function WeatherWidget({ location }: Props) {
     };
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
     fetchForecast();
-  }, [location, enqueueSnackbar]);
+  }, [location, unit, enqueueSnackbar]);
 
-  const roundTemperature = (temp: number) => Math.floor(temp / 10);
+  const roundTemperature = (temp: number) => Math.floor(temp);
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: why you removed it?

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 6]
```diff
@@ -32,12 +32,13 @@ type Forecast = Array<[string, ForecastBody[]]>;
 
 interface Props {
   location: Location;
+  unit: string;
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: Also make me stricter

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 7]
```diff
@@ -124,9 +126,16 @@ export default function WeatherWidget({ location }: Props) {
                         alt='Weather condition'
                       />
                     </Tooltip>
-                    {`Temperature: ${roundTemperature(
-                      temp,
-                    )}°C Humidity: ${humidity}%`}
+                    {`Temperature: ${roundTemperature(temp)}${(() => {
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: Extract the function that pretty-prints temperature units out of markup similar to `roundUnits`

File: src/components/WeatherWidget/WeatherWidget.tsx (hunk 5 above)
Comment (coding-vibe) [reply src/components/WeatherWidget/WeatherWidget.tsx]: I don't know why, but now it is working without it🤷‍♀️

File: src/utils/units.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+const UNITS = ['standard', 'metric', 'imperial'];
```
Comment (singvarr) [inline src/utils/units.ts]: Put me in enum. Also this file should be in `constants` folder

File: src/App.tsx [hunk 9]
```diff
@@ -1,12 +1,16 @@
...
+import TemperatureUnitsSelect from 'components/TemperatureUnitsSelect';
 import Location from 'types/location';
+import Unit from 'types/unit';
 
 const LOCATION_AUTOCOMPLETE = 'location-select';
+const TEMPERATURE_UNITS_SELECT = 'unit-select';
 
 function App() {
   const [selectedLocation, onSelectLocation] = useState<Location | null>(null);
+  const [selectedUnit, onSelectUnit] = useState<Unit>('metric');
```
Comment (singvarr) [inline src/App.tsx]: Use value from enum

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 10]
```diff
@@ -77,9 +80,19 @@ export default function WeatherWidget({ location }: Props) {
...
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
     fetchForecast();
-  }, [location, enqueueSnackbar]);
+  }, [location, unit, enqueueSnackbar]);
 
-  const roundTemperature = (temp: number) => Math.floor(temp / 10);
+  const roundTemperature = (temp: number) => Math.floor(temp);
+  const combineUnitWithSymbol = (tempUnit: Unit) => {
+    switch (tempUnit) {
+      case 'metric':
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: Use value from enum

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 11]
```diff
@@ -77,9 +80,19 @@ export default function WeatherWidget({ location }: Props) {
...
+        return '\u2103';
+      case 'imperial':
+        return '\u00B0F';
+      default:
+        return '\u00B0K';
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: Move Kelvin to a case, raise exception in `default`

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 12]
```diff
@@ -77,9 +80,19 @@ export default function WeatherWidget({ location }: Props) {
     };
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
     fetchForecast();
-  }, [location, enqueueSnackbar]);
+  }, [location, unit, enqueueSnackbar]);
 
-  const roundTemperature = (temp: number) => Math.floor(temp / 10);
+  const roundTemperature = (temp: number) => Math.floor(temp);
+  const combineUnitWithSymbol = (tempUnit: Unit) => {
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: `formatTemperatureUnits`

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 13]
```diff
@@ -126,7 +139,8 @@ export default function WeatherWidget({ location }: Props) {
                     </Tooltip>
                     {`Temperature: ${roundTemperature(
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: Let's create a function that receives a number with temparature and returns pretty-printed value

File: src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx [hunk 14]
```diff
@@ -0,0 +1,36 @@
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import Unit from 'types/unit';
+import UNITS from 'utils/units';
+import * as classes from './styles';
+
+interface Props {
+  unit: Unit;
```
Comment (singvarr) [inline src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx]: rename everywhere in the project unit to `temperatureUnit`

File: src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+  setUnit: (value: Unit) => void;
+  id: string;
+}
+
+export default function TemperatureUnitsSelect({ unit, setUnit, id }: Props) {
+  const [standard, metric, imperial] = UNITS;
+  const handleChange = (event: SelectChangeEvent) => {
```
Comment (singvarr) [inline src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx]: Add generic for `SelectChangeEvent`

File: src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+  return (
+    <FormControl css={classes.select}>
+      <InputLabel id='helper-label'>Temperature unit</InputLabel>
+      <Select
+        labelId='helper-label'
+        id={id}
+        value={unit}
+        label='Temperature unit'
+        onChange={handleChange}>
+        <MenuItem value={standard}>Kelvin, &deg;&#8490;</MenuItem>
```
Comment (singvarr) [inline src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx]: Can be mapped. Create another array of object that keeps a value from enum and label for select

File: src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+
+export default function TemperatureUnitsSelect({ unit, setUnit, id }: Props) {
+  const [standard, metric, imperial] = UNITS;
+  const handleChange = (event: SelectChangeEvent) => {
+    setUnit(event.target.value);
+  };
+
+  return (
+    <FormControl css={classes.select}>
+      <InputLabel id='helper-label'>Temperature unit</InputLabel>
```
Comment (singvarr) [inline src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx]: Make label more precise

File: src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+import TemperatureUnits from 'constants/temperatureUnits';
+
+interface Props {
+  temperatureUnit: TemperatureUnits;
+  setTemperatureUnit: (value: TemperatureUnits) => void;
+  id: string;
+  className?: string;
+}
+
+const temperatureUnitsOptions = [
```
Comment (singvarr) [inline src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx]: Move me to a separate file inside this folder

File: .eslintrc.cjs [hunk 19]
```diff
@@ -62,6 +62,8 @@ module.exports = {
     'prettier/prettier': 2,
     "no-unused-vars": "off",
     "@typescript-eslint/no-unused-vars": ["error"],
+    "no-shadow": "off",
```
Comment (singvarr) [inline .eslintrc.cjs:65]: Why?

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 20]
```diff
@@ -77,9 +80,23 @@ export default function WeatherWidget({ location }: Props) {
     };
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
     fetchForecast();
-  }, [location, enqueueSnackbar]);
+  }, [location, temperatureUnit, enqueueSnackbar]);
 
-  const roundTemperature = (temp: number) => Math.floor(temp / 10);
+  const formatTemperatureUnits = (tempUnit: TemperatureUnits) => {
+    const { KELVIN, CELSIUS, FAHRENHEIT } = TemperatureUnits;
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: Don't destructure enum, please

File: src/components/WeatherWidget/WeatherWidget.tsx [hunk 21]
```diff
@@ -124,9 +141,11 @@ export default function WeatherWidget({ location }: Props) {
                         alt='Weather condition'
                       />
                     </Tooltip>
-                    {`Temperature: ${roundTemperature(
+                    {`Temperature: ${formatTemperatureData(
                       temp,
-                    )}°C Humidity: ${humidity}%`}
+                      formatTemperatureUnits(temperatureUnit),
```
Comment (singvarr) [inline src/components/WeatherWidget/WeatherWidget.tsx]: redundant second argument

File: src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+            value={option.value}>
+            {option.name}
+          </MenuItem>
+        ))}
+      </Select>
+    </FormControl>
+  );
+}
+
+TemperatureUnitsSelect.defaultProps = {
```
Comment (singvarr) [inline src/components/TemperatureUnitsSelect/TemperatureUnitsSelect.tsx:46]: Add default value to function argument instead

**Review by singvarr — APPROVED** [review]
LGTM. We have a couple of final codestyle issues and we can merge it after its fix

**Pull request merged**